sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -135,7 +133,9 @@ object ScalaNativePluginInternal {
         case Some(_) =>
           ()
       }
-    }
+    },
+    nativeGC := "boehm",
+    nativeGC in NativeTest := (nativeGC in Test).value
```

Comment:
`nativeGC` should be in base settings, not global settings. Global settings are not scoped separately for compile and test and should generally be avoided.

Reply:
What's the concern with global settings not being scoped separately for compile and test?

I believe the intent here is to define the default value of for the key `nativeGC` (introduced by this plugin), which is one of the use cases for which global settings _should_ be used (http://www.scala-sbt.org/1.0/docs/Plugins.html#projectSettings+and+buildSettings).

The benefit of doing this is you if you want to redefine the value of `nativeGC` for all the projects in your build you can do so by defining `nativeGC in ThisBuild := "immix"` (either in your build code, or in the shell with `set`).

Reply:
I was not able to get a behaviour where I could change the `nativeGC` setting for ScalaNative or a user project that would apply across all the native projects so this is what led me to the global key along with help from Dale. The original intent of this PR was just to be able to `set nativeGC in ThisBuild := "immix"` and all the other changes are somewhat incidental to making that happen. I really tried to make as few changes as possible.